src/pyk/proof/reachability.py

```diff
@@ -480,6 +489,9 @@ def read_proof_data(proof_dir: Path, id: str) -> APRProof:
         node_refutations = {
             kcfg._resolve(int(node_id)): proof_id for node_id, proof_id in proof_dict['node_refutations'].items()
         }
+        prior_loops_cache = {
+            loop_node_id: list(same_loops) for (loop_node_id, same_loops) in proof_dict['loops_cache'].items()
+        }
```

Comment:
This will cause a breakage in reading existing proofs from disk because they won't have the `loops_cache` key. Also, we can just use the dictionary directly, instead of doing any processing on it I think:

```suggestion
        prior_loops_cache = proof_dict.get('loops_cache', {})
```